Remove 9.6

jasonmp85 opened the pull request:
DESCRIPTION: Removes support and code for PostgreSQL 9.6

mtuncer commented:
Can we continue with this work ? Travis is running extra PG 9.6 for each commit. Could we at least remove 9.6 from build matrix and merge that change without waiting for the rest of the changes ?

marcocitus commented:
> Can we continue with this work ? Travis is running extra PG 9.6 for each commit. Could we at least remove 9.6 from build matrix and merge that change without waiting for the rest of the changes ?

I'd be ok with pushing https://github.com/citusdata/citus/pull/2554/commits/34e495a8e090afa4a3408bbb439753de576b6d59 straight to master.

jasonmp85 commented:
> Can we continue with this work?

Yes. I'm back from winter break and plan on addressing all feedback today.

metdos commented:
> Yes. I'm back from winter break and plan on addressing all feedback today.

@jasonmp85 any update on this?

jasonmp85 commented on src/backend/distributed/commands/create_distributed_table.c:
```diff
@@ -1191,11 +1191,7 @@ RegularTable(Oid relationId)
 {
```
Updated it.

jasonmp85 commented on src/backend/distributed/commands/create_distributed_table.c:
```diff
@@ -1388,7 +1384,6 @@ TupleDescColumnNameList(TupleDesc tupleDescriptor)
 static bool
```
Keeping this one, but changed _supported_ to _introduced in_.

jasonmp85 commented on src/backend/distributed/commands/schema.c:
```diff
@@ -147,13 +147,8 @@ GetSchemaNameFromDropObject(ListCell *dropSchemaCell)
 {
```
Good catch. I've now done precisely that (removed it).

jasonmp85 commented on src/backend/distributed/commands/table.c:
```diff
@@ -129,7 +129,6 @@ ProcessDropTableStmt(DropStmt *dropTableStatement)
 void
```
Fixed.

jasonmp85 commented on src/backend/distributed/commands/table.c:
```diff
@@ -194,7 +192,6 @@ ProcessCreateTableStmtPartitionOf(CreateStmt *createStatement)
 void
```
(In the future, leave GitHub comments _on the line in question_. Putting them on the `void` line like this means I don't see the comment at all in GitHub without expanding some block or other in the UI).

Fixed.

jasonmp85 commented on src/backend/distributed/commands/transmit.c:
```diff
@@ -107,25 +103,16 @@ SendRegularFile(const char *filename)
 
```
If something mentions Hadoop, it is _quite_ stale indeed.

Marking as resolved as there's no action for me here.

jasonmp85 commented on src/backend/distributed/commands/utility_hook.c:
```diff
@@ -96,16 +74,12 @@ void
 CitusProcessUtility(Node *node, const char *queryString, ProcessUtilityContext context,
```
Changed the comment to describe it as a convenience method to wrap arguments in a `PlannedStmt` before passing along `ProcessUtility`.

jasonmp85 commented on src/backend/distributed/connection/remote_commands.c:
```diff
@@ -824,13 +800,8 @@ WaitForAllConnections(List *connectionList, bool raiseInterrupts)
 			}
```
PLEASE comment _on the line you're talking about_. This is getting very confusing.

jasonmp85 commented on src/backend/distributed/transaction/citus_dist_stat_activity.c:
```diff
@@ -125,149 +123,75 @@
  * Also, backend_type introduced with pg 10+ so we have null in the previous verions.
  */
 
-#if PG_VERSION_NUM >= 100000
```
Fixed.

jasonmp85 commented:
> @jasonmp85 any update on this?

@metdos I just pushed a commit addressing all feedback; sorry, was a bit delayed due to home office move, screwed up laptop workspace (long story), and CircleCI work.

Since all of the feedback was on comments, I'm guessing the code was fine… going to merge this once it passes. @velioglu: if you _did_ have more you wanted feedback on, let me know. Since this is such a big patch, I don't want it sitting around any longer, though.

jasonmp85 commented:
Removes support and code for PostgreSQL 9.6
